scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3800,7 +3831,11 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
 
   def enumSingleCaseDef(mods: List[Mod]): Defn.EnumCase = {
     val name = termName()
-    val tparams = typeParamClauseOpt(ownerIsType = true, ctxBoundsAllowed = true)
+    val tparams = typeParamClauseOpt(
+      ownerIsType = true,
+      ctxBoundsAllowed = true,
+      allowUnderscore = dialect.allowTypeParamUnderscore
```

Comment:
Why not have it in one place only? Why do we need to forward it through all the functions?

Reply:
That's because it is not in all functions, even example in issue shows this:
`F[_]` <- should fail
`F[X[_]]` <- valid
and this function is called recursively.

Reply:
Ok, thanks for the explanation!